src/Fixer/PhpUnit/PhpUnitAttributesFixer.php

```diff
@@ -45,29 +62,29 @@ public function __construct()
 
     public function getDefinition(): FixerDefinitionInterface
     {
+        $codeSample = <<<'PHP'
+            <?php
+            /**
+             * @covers \VendorName\Foo
+             * @internal
+             */
+            final class FooTest extends TestCase {
+                /**
+                 * @param int $expected
+                 * @param int $actual
+                 * @dataProvider giveMeSomeData
+                 * @requires PHP 8.0
+                 */
+                public function testSomething($expected, $actual) {}
+            }
+
+            PHP;
+
         return new FixerDefinition(
             'PHPUnit attributes must be used over their respective PHPDoc-based annotations.',
             [
-                new VersionSpecificCodeSample(
-                    <<<'PHP'
-                        <?php
-                        /**
-                         * @covers \VendorName\Foo
-                         * @internal
-                         */
-                        final class FooTest extends TestCase {
-                            /**
-                             * @param int $expected
-                             * @param int $actual
-                             * @dataProvider giveMeSomeData
-                             * @requires PHP 8.0
-                             */
-                            public function testSomething($expected, $actual) {}
-                        }
-
-                        PHP,
-                    new VersionSpecification(8_00_00),
-                ),
+                new VersionSpecificCodeSample($codeSample, new VersionSpecification(8_00_00)),
+                new VersionSpecificCodeSample($codeSample, new VersionSpecification(8_00_00), ['keep_annotations' => true]),
```

Comment:
```suggestion
                new VersionSpecificCodeSample($codeSample, new VersionSpecification(8_00_00), ['keep_annotations' => true]),
                new VersionSpecificCodeSample($codeSample2, new VersionSpecification(8_00_00), ['keep_annotations' => false]),
```

I would expect annotations are removed in new sample

Reply:
The whole point of introducing `$codeSample` was to reuse the code snippet. I don't see the need for different code to show different behaviour, so I would keep it as is. If we need different code snippet for examples, then let's get rid of interim variables, just put the snippet in the `VersionSpecificCodeSample`.

Reply:
i don't mind about having or not having the var for it.

i mind to have the sample that is showing that if both, annotation and attribute are in place, rule can remove annotation

Reply:
i don't mind about having or not having the var for it.

i mind to have the sample that is showing that if both, annotation and attribute are in place, rule can remove annotation, as current samples are showing that rule is only adding sth

Reply:
Not really. Example 1 removes annotations and adds attributes, while example 2 reuses the same code snippet, but only adds attributes, keeping the annotations. IMHO these 2 examples show exactly what they should.

Reply:
`@` = annotation
`#` = attributes

we have samles for:
`@` -> `#`
`@` -> `@#` (for new config option, when ppl want to use both, for interim stage of supporting both versions of PHPUnit)

what we miss is:
`@#` -> `#` (when ppl no longer need both and can move fully to `#`)

(again, i don't mind if we reuse the same code snippet/variable/etc)

Reply:
Ah OK, now I understand. @kubawerlos so we can use the same snippet as is now (for 2 examples), but we need another example showing that annotations can be removed even if attribute is already there (not added during fix). Makes sense 👍.

Reply:
> I would expect annotations are removed in new sample

The new sample with configuration `['keep_annotations' => true]`?

> what we miss is: `@#` -> `#` (when ppl no longer need both and can move fully to `#`)

Yes, we miss that functionality, so I'd say it is expected that there is no code sample for it.

Reply:
would be nice to have it. otherwise we provide way to migrate step by step, but we give only half of steps :D 
ok to go separately 👍🏻

Reply:
I think follow-up ticket was not created, let me make it:
https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/8100

if already existing, please close newly created one as duplicate